src/Plugin/SmsGateway/Twilio.php

```diff
@@ -145,4 +152,64 @@ public function send(SmsMessageInterface $sms_message) {
     return $result;
   }
 
+  /**
+   * @inheritDoc
+   */
+  public function incoming(SmsMessageInterface $sms_message) {
```

Comment:
It's required for the Plugin to implement `SmsGatewayPluginIncomingInterface`. The logic provided is pretty basic and I'm adding a test on the data being stored in the "data" option for setting the status... but we have to implement it on the plugin. SMS module tests for that interface being implemented to determine if the plugin can handle incoming SMS.